pkg/rancher/migrations.go

```diff
@@ -273,6 +280,68 @@ func addWebhookConfigToCAPICRDs(crdClient controllerapiextv1.CustomResourceDefin
 	return nil
 }
 
+func migrateAddCAPIWatchFilterLabels(w *wrangler.Context) error {
+	namespaces, err := w.Core.Namespace().List(metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("listing namespaces: %w", err)
+	}
+	for _, ns := range namespaces.Items {
+		// Clusters
+		clusters, err := w.CAPI.Cluster().List(ns.Name, metav1.ListOptions{})
+		if err != nil {
+			return fmt.Errorf("listing capi clusters in ns %s: %w", ns.Name, err)
+		}
+		for _, cluster := range clusters.Items {
+			if isRancherOwned(cluster.OwnerReferences) {
+				if _, ok := cluster.Labels[capi.WatchLabel]; !ok {
+					clusterCopy := cluster.DeepCopy()
+					clusterCopy.Labels[capi.WatchLabel] = rke2.CAPIFilterValue
+					if _, updateErr := w.CAPI.Cluster().Update(clusterCopy); updateErr != nil {
+						return fmt.Errorf("saving update to cluster %s: %w", clusterCopy.Name, updateErr)
+					}
+				}
+			}
+		}
+
+		// MachineDeployments
```

Comment:
do we have to update machinesets too? They can be created as stand-alone resource